File: backend/RentLoop/RentLoop.API/DTOs/Reservations/ReservationCreateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace RentLoop.API.DTOs.Reservations
{
    public class ReservationCreateRequest
    {
        public int ListingId { get; set; }

        public DateTime CheckIn { get; set; }
        public DateTime CheckOut { get; set; }

        public int Guests { get; set; }

        [MaxLength(1000)]
        public string? Note { get; set; }
    }
}

File: frontend/rentloop_app/lib/features/user/models/reservation_create_request.dart
class ReservationCreateRequest {
  final int listingId;
  final DateTime checkIn;
  final DateTime checkOut;
  final int guests;
  final String? note;

  ReservationCreateRequest({
    required this.listingId,
    required this.checkIn,
    required this.checkOut,
    required this.guests,
    this.note,
  });

  Map<String, dynamic> toJson() => {
        'listingId': listingId,
        'checkIn': checkIn.toUtc().toIso8601String(),
        'checkOut': checkOut.toUtc().toIso8601String(),
        'guests': guests,
        'note': note,
      };
}
